indexer/httpd/src/graphql/query/message.rs

```diff
@@ -0,0 +1,184 @@
+        let app_ctx = ctx.data::<Context>()?;
+
+        query_with(
+            after,
+            before,
+            first,
+            last,
+            |after: Option<MessageCursorType>, before: Option<MessageCursorType>, first, last| async move {
+                let mut query = entity::messages::Entity::find();
+                let sort_by = sort_by.unwrap_or_default();
+                let limit;
+                let has_before = before.is_some();
+
+                match (after, before, first, last) {
+                    (after, None, first, None) => {
+                        if let Some(after) = after {
+                            query = apply_filter(query, sort_by, &after);
+                        }
+
+                        limit = first.map(|x| x as u64).unwrap_or(MAX_MESSAGES);
+
+                        query = query.limit(limit + 1);
+                    }
+                    (None, before, None, last) => {
+                        if let Some(before) = before {
+                            query = apply_filter(query, sort_by, &before);
+                        }
+
+                        limit = last.map(|x| x as u64).unwrap_or(MAX_MESSAGES);
+
+                        query = query.limit(limit + 1);
+                    }
+                    _ => unreachable!()
+                }
+
+                if let Some(block_height) = block_height {
+                    query = query.filter(entity::messages::Column::BlockHeight.eq(block_height as i64));
+                }
+
+                if let Some(method_name) = method_name {
+                    query = query.filter(entity::messages::Column::MethodName.eq(method_name));
+                }
+
+                if let Some(contract_addr) = contract_addr {
+                    query = query.filter(entity::messages::Column::ContractAddr.eq(contract_addr));
+                }
+
+                if let Some(sender_addr) = sender_addr {
+                    query = query.filter(entity::messages::Column::SenderAddr.eq(sender_addr));
+                }
+
+                match sort_by {
+                    SortBy::BlockHeightAsc => {
+                        query = query
+                            .order_by(entity::messages::Column::BlockHeight, Order::Asc)
+                            .order_by(entity::messages::Column::OrderIdx, Order::Asc)
+                    },
+                    SortBy::BlockHeightDesc => {
+                        query = query
+                            .order_by(entity::messages::Column::BlockHeight, Order::Desc)
+                            .order_by(entity::messages::Column::OrderIdx, Order::Desc)
+                    },
+                }
+
+                let mut messages: Vec<types::message::Message> = query
+                .all(&app_ctx.db)
+                .await?
+                .into_iter()
+                .map(|message| message.into())
+                .collect::<Vec<_>>();
```

Comment:
For some reason, rustfmt doesn't work in some of the files. For example you can see clearly this part isn't formatted correctly